src/Mod/Part/App/Geometry.cpp

```diff
@@ -1516,6 +1516,37 @@ bool GeomBSplineCurve::removeKnot(int index, int multiplicity, double tolerance)
     }
 }
 
+void GeomBSplineCurve::Trim(double u, double v)
+{
+    auto splitUnwrappedBSpline = [this](double u, double v) {
+        // it makes a copy internally (checked in the source code of OCCT)
+        auto handle = GeomConvert::SplitBSplineCurve (  myCurve,
+                                                            u,
+                                                            v,
+                                                            Precision::Confusion()
+                                                        );
+        setHandle(handle);
+    };
+
+    try {
+        if(!isPeriodic()) {
+            splitUnwrappedBSpline(u, v);
+        }
+        else { // periodic
+            if( v < u ) { // wraps over origin
+                v = v + 1.0; // v needs one extra lap (1.0)
+
+                splitUnwrappedBSpline(u, v);
+            }
+            else {
+                splitUnwrappedBSpline(u, v);
+            }
+        }

```

Comment:
Yes. It makes sense. 

At the beginning there was a lot of other code because I thought I needed to move the origin (the seam) of the periodic b-spline. Finally, looking at the OCCT code I realise that adding a lap would work, I just removed the superseeded code.